stan/math/prim/fun/gaus_interp.hpp

```diff
@@ -0,0 +1,202 @@
+#ifndef STAN_MATH_PRIM_FUN_GAUS_INTERP
+#define STAN_MATH_PRIM_FUN_GAUS_INTERP
+
+#include <stan/math/prim/fun/conv_gaus_line.hpp>
+#include <stan/math/prim/fun/square.hpp>
+#include <stan/math/prim/err.hpp>
+#include <cmath>
+#include <vector>
+#include <algorithm>
+
+namespace stan {
+namespace math {
+
+// struct for passing precomputation data
+struct gaus_interp_params {
+  std::vector<double> as;
+  std::vector<double> bs;
+  double sig2;
+};
```

Comment:
I got rid of the gaus_interp_params struct altogether and instead just used a vector<double> to store the parameters. I think we discussed doing this because otherwise we would have had to create a new data type in Stan for this struct.